### ISSUE TITLE
CMD.exe terminal no longer substitutes environment variables in its PATH

### ISSUE THREAD

**Issue by TheJCAB** [microsoft/vscode#233367]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version:
Version: 1.95.1 (user setup)
Commit: 65edc4939843c90c34d61f4ce11704f09d3e5cb6
Date: 2024-10-31T05:14:54.222Z
Electron: 32.2.1
ElectronBuildId: 10427718
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
- OS Version: 
OS: Windows_NT x64 10.0.26100

Steps to Reproduce:

1. Open a CMD.exe terminal in VS Code (on Windows).
2. Run the command "set PATH"
3. Observe:

```
C:\TEMP>set path
Path=C:\Program Files\...;<multiple paths>;...\bin;%USERPROFILE%\AppData\Local\Microsoft\WindowsApps;%USERPROFILE%\.dotnet\tools;...;<more paths>
```

Normal command windows, whether using the old terminal window or the new one, elevated or not, show those %USERPROFILE% variables in the PATH expanded and replaced with their values.

The result is that command-line tools installed into "%USERPROFILE%\AppData\Local\Microsoft\WindowsApps;" no longer run from within the VS Code terminal window. See for instance, I can no longer invoke the WinDbg debugger from the VSCode terminal:

```
C:\TEMP>windbgx
'windbgx' is not recognized as an internal or external command,
operable program or batch file.
```

This works fine in all other command windows in my machine, and used to work from within VSCode.

This seems to be a regression (I recently allowed VS Code to update itself).

**Comment by meganrogge** [comment]
nothing has changed recently from our side. if you install earlier versions, does it work?

**Comment by TheJCAB** [comment]
I don't know what to say. I run "windbgx" regularly from the terminal window, and as soon as I got the update it stopped working. And it only happens to terminal windows in VSCode, AFACS. regular desktop console windows don't have the problem. I might not have updated to every intermediate version (I've been known to ignore the prompts :-)), and I don't know what I had before.

Is it possible to install old versions side-by-side? I can try to bisect for you.

**Comment by meganrogge** [comment]
This type of issue is typically related to environment differences between VS Code's terminal and the external terminal. You can try diffing the two environments by printing them (eg. `env`, `Get-ChildItem env:`) and investigating any differences.

Closing this off as it doesn't look actionable on our end, you can read more about our [issue process here](https://github.com/microsoft/vscode/wiki/Terminal-Issues#why-did-you-close-my-issue). Thanks for understanding.

**Comment by meganrogge** [comment]
If this is a confirmed regression, it would help to know which change caused it. We maintain a node.js based CLI tool  [`vscode-bisect`](https://github.com/microsoft/vscode-bisect) that automatically downloads and runs previous VS Code insiders and asks for each build if the issue reproduces. It takes up to 8 steps to find exactly the build that caused the regression.

**Steps:**
* install [node.js](https://nodejs.org/en/download) (if not yet done)
* `npx --yes @vscode/vscode-bisect@latest`
* follow the instructions until you found the offending build (its fine to leave the commit empty)
* report back the commit range in this issue

Thanks!

**Comment by TheJCAB** [comment]
I can verify that it's not happening anymore. I seem to have been upgraded to 1.95.3 :-P. And I've been made to reboot my computer a couple of times since.

"try diffing the two environments" -- The issue, as I reported, is the difference in the environment. Inside of VSCode's terminal, variables like "%USERPROFILE%" in environment variables (especially PATH) were not expanded, but outside of VSCode they were.

I always open VSCode using the Windows UI (not from the command-line), and in fact, all instances I use are invariably re-opened as long as I click the VSCode icon that I have pinned in my taskbar. I had been seeing other subtly strange things on Windows, all of which seem to be resolved after the reboots, so I have to assume Windows was the culprit. Maybe all UI apps were getting bad environments.

Anyway, I agree this item should be closed now.

Thanx!
